### PR TITLE
feat(storage): add support for Elasticsearch 8

### PR DESCRIPTION
**Pull request by rolljee**
## What does this PR do ?

- This PR introduce the support of Elasticsearch 8

The choice made is to keep kuzzle compatible to avoid breaking changes around the support of ES8.

We wanted to allow the user to OPT-IN for the feature. By default the majorVersion support will be 7 until Kuzzle v3.

So now, and new key is available under

```json
{
  "services": {
    "storageEngine": {
      "majorVersion": 8
    }
  }
}
```

This key will load the wanted sdk version and will allow the user to use the new features of ES8.

[ ] - Documentation
[ ] - tests

### How should this be manually tested?

- Step 1 : Play with the majorVersion key
- Step 2 : docker compose up -d

### Other changes

- Linting

### CODE REVIEW

File: doc/2/guides/elasticsearch/v8/index.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  "services": {
+    "storageEngine": {
+      "majorVersion": 8
+    }
+  }
+}
+```
+
+:::warning
+You can not set the majorVersion to 8 if you are using a version of Kuzzle that does not support it.
```
Comment (Kuruyia) [inline doc/2/guides/elasticsearch/v8/index.md:44]:
```suggestion
You can not set the `majorVersion` key to 8 if you are using a version of Kuzzle that does not support it.
```

File: doc/2/guides/elasticsearch/v8/index.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+:::
+
+:::info
+Kuzzle cannot connect to both ES7 and ES8 at the same time.
```
Comment (Kuruyia) [inline doc/2/guides/elasticsearch/v8/index.md:48]:
```suggestion
Kuzzle cannot connect to both Elasticsearch 7 and Elasticsearch 8 at the same time.
```

**Review by Kuruyia — CHANGES_REQUESTED** [review]
Note: we might want to update the copyright dates in the source files (in a separate PR).

**Review by etrousset — APPROVED** [review]

File: doc/2/guides/elasticsearch/v8/index.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+}
+```
+
+:::warning
+You can not set the `majorVersion` key to 8 if you are using a version of Kuzzle that does not support it. 
+:::
+
+:::info
+Kuzzle cannot connect to both Elasticsearch 7 and Elasticsearch 8 at the same time.  
+:::
```
Comment (afondard) [inline doc/2/guides/elasticsearch/v8/index.md:49]: I guess it's not ::: but ``` here on these 7 lines

File: doc/2/guides/elasticsearch/v8/index.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+	* Step 1: Disable Shard Allocation.
+	* Step 2: Stop and upgrade a single Elasticsearch node.
+	* Step 3: Enable Shard Allocation and allow the node to join the cluster and the cluster to re-balance.
+	* Step 4: Repeat for each node in the cluster.
+1. Post Upgrade Checks
+	* Run the health and stats APIs to ensure the health of your newly upgraded cluster.
+	* Update your clients and integrations to the latest version that's compatible with Elasticsearch 8.x, if not done already.
+	* Monitor your cluster using the Monitoring API or third-party monitoring services.
+1. Troubleshoot
+  * If you encounter any issues during the migration process, take advantage of the Elasticsearch documentation, forums, and issue trackers for troubleshooting information and support.
```
Comment (afondard) [inline doc/2/guides/elasticsearch/v8/index.md:152]: Align bullet point with the others

```suggestion
	* If you encounter any issues during the migration process, take advantage of the Elasticsearch documentation, forums, and issue trackers for troubleshooting information and support.
```

**Review by afondard — APPROVED** [review]

File: doc/2/guides/elasticsearch/v8/index.md (hunk 3 above)
Comment (rolljee) [reply doc/2/guides/elasticsearch/v8/index.md:49]: This is part of the kuzzle documentation paradigm :)

**Review by alexandrebouthinon — APPROVED** [review]

**Comment by rolljee** [comment]
Elasticsearch-8 branch has been created, work will continue from there.

Thanks everyone that has participated in this.

**Pull request closed without merging**